server/searchHandler.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/go-github/github"
```

Comment:
In process of scoping out all the functionality I'd have to write to support Github's API, came across this which did everything (and more) that I was looking for (auth, search API support, pagination, rate limits). Has fairly good tests, large active community, actively maintained.